Makefile

```
@@ -298,84 +305,84 @@ $(O_FILES): | asset_files
 
 .PHONY: o_files asset_files
 
-build/$(SPEC): $(SPEC)
-	$(CPP) $(CPPFLAGS) $< > $@
+$(BUILD_DIR)/$(SPEC): $(SPEC)
+	$(CPP) $(CPPFLAGS) $< | sed -e 's|$$(BUILD_DIR)|$(BUILD_DIR)|g' > $@
```

```suggestion
	$(CPP) $(CPPFLAGS) $< | $(SPEC_REPLACE_TOKENS) > $@
```
`SPEC_REPLACE_TOKENS := sed -e 's|$$(BUILD_DIR)|$(BUILD_DIR)|g'`
Does something like this work? I think it could be nice to use this to consolidate the two cases where this is used.
(Also if you think there's a better name feel free to use it)

---

Okay, I made `SPEC_REPLACE_VARS`. I didn't go all the way because I think it's a bit strange to write a file when reading the Makefile instead of while executing it